plugins/newsdownloader.koplugin/epubdownloadbackend.lua

```diff
@@ -353,10 +353,9 @@ function EpubDownloadBackend:createEpub(epub_path, html, url, include_images, me
                 src_ext = src_ext:match("(.-)%?") -- remove ?blah
             end
             local ext = src_ext:match(".*%.(%S%S%S?%S?%S?)$") -- extensions are only 2 to 5 chars
-            if ext == nil or ext == "" then
-                -- we won't know what mimetype to use, ignore it
-                logger.dbg("no file extension found in ", src)
-                return nil
+            if ext == nil then
+                --- @todo Reverse the logic to download the image first so we can get the mimetype from the headers?
+                ext = ""
```

Comment:
Or may be force it to be "gif" (crengine and probably other readers won't care if it's the truth or a lie), to avoid this ugly:
`2024.epub/OEBPS/images/img00001.(-1,-1): Filename is not allowed to end with ".".`

Reply:
For my own sanity it's better to just skip the dot (see new commit). :-)

Reply:
That results in (none of which new):

```
epubcheck '/home/frans/src/ko/koreader/koreader-emulator-x86_64-linux-gnu-debug/koreader/news/Betty/24-12-31_01-00_Betty by Gary Delainey and Gerry Rasmussen for Tue_ 31 Dec 2024.epub' 
Validating using EPUB version 2.0.1 rules.
ERROR(PKG-007): ./24-12-31_01-00_Betty by Gary Delainey and Gerry Rasmussen for Tue_ 31 Dec 2024.epub/mimetype(-1,-1): Mimetype file should only contain the string "application/epub+zip" and should not be compressed.
ERROR(RSC-005): ./24-12-31_01-00_Betty by Gary Delainey and Gerry Rasmussen for Tue_ 31 Dec 2024.epub/OEBPS/content.opf(9,14): Error while parsing file: element "metadata" incomplete; missing required elements "dc:identifier" and "dc:language"
ERROR(OPF-030): ./24-12-31_01-00_Betty by Gary Delainey and Gerry Rasmussen for Tue_ 31 Dec 2024.epub/OEBPS/content.opf(-1,-1): The unique-identifier "bookid" was not found.
ERROR(HTM-004): ./24-12-31_01-00_Betty by Gary Delainey and Gerry Rasmussen for Tue_ 31 Dec 2024.epub/OEBPS/content.html(-1,-1): Irregular DOCTYPE: found "", expected "<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1//EN" "http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd">".
ERROR(RSC-005): ./24-12-31_01-00_Betty by Gary Delainey and Gerry Rasmussen for Tue_ 31 Dec 2024.epub/OEBPS/content.html(2,7): Error while parsing file: elements from namespace "" are not allowed
FATAL(RSC-016): ./24-12-31_01-00_Betty by Gary Delainey and Gerry Rasmussen for Tue_ 31 Dec 2024.epub/OEBPS/content.html(3,109): Fatal Error while parsing file: The element type "meta" must be terminated by the matching end-tag "</meta>".
ERROR(RSC-005): ./24-12-31_01-00_Betty by Gary Delainey and Gerry Rasmussen for Tue_ 31 Dec 2024.epub/OEBPS/content.html(-1,-1): Error while parsing file: The element type "meta" must be terminated by the matching end-tag "</meta>".
ERROR(HTM-049): ./24-12-31_01-00_Betty by Gary Delainey and Gerry Rasmussen for Tue_ 31 Dec 2024.epub/OEBPS/content.html(2,7): Html element does not have an xmlns set to "http://www.w3.org/1999/xhtml".

Check finished with errors
Messages: 1 fatal / 7 errors / 0 warnings / 0 infos

EPUBCheck completed
```